src/benchmarks/micro/libraries/System.Text.Json/Serializer/ReadMissingAndCaseInsensitive.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using MicroBenchmarks.Serializers;
+using Newtonsoft.Json.Serialization;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    [GenericTypeArguments(typeof(Location))]
+    public class ReadMissingAndCaseInsensitive<T>
+    {
+        private string _serialized;
+        private string _serializedCamelCased;
+        private JsonSerializerOptions _optionsBaseline;
+        private JsonSerializerOptions _optionsCaseInsensitive;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            T value = DataGenerator.Generate<T>();
+            _serialized = JsonSerializer.Serialize(value);
+
+            JsonSerializerOptions options = new JsonSerializerOptions()
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+            };
+
+            _serializedCamelCased = JsonSerializer.Serialize(value, options);
+
+            _optionsBaseline = new JsonSerializerOptions();
+
+            _optionsCaseInsensitive = new JsonSerializerOptions
+            {
+                PropertyNameCaseInsensitive = true
+            };
+        }
+
+        [BenchmarkCategory(Categories.Libraries, Categories.JSON)]
+        [Benchmark]
+        public T DeserializeBaseline() => JsonSerializer.Deserialize<T>(_serialized, _optionsBaseline);
+
+        [BenchmarkCategory(Categories.Libraries, Categories.JSON)]
+        [Benchmark]
+        public T DeserializeMissing() => JsonSerializer.Deserialize<T>(_serializedCamelCased, _optionsBaseline);
```

Comment:
Can you rename this? The current name took going to your original issue for me to understand.
```suggestion
        public T DeserializeWithMissingProperties() => JsonSerializer.Deserialize<T>(_serializedCamelCased, _optionsBaseline);
```

May also be worth a comment to make it clear that the properties will be "missing" because we're not using the case insensitive option.

Reply:
Hmm one issue with names is that the results comparer truncates at 80 chars, so I've been trying to keep names small otherwise it is not possible to tell in the output.

However I didn't do a good job at that. It looks like `System.Text.Json.Serialization.Tests.ReadMissingAndCaseInsensitive<Location>.` is already at 77 characters leaving only 3 characters which currently is not even enough to distinguish between the 4 tests I added.

We could fix that issue with the results comparer, or I could simplify the class name and remove "DeserializeWith" but that still doesn't leave enough room to add "Properties".

Reply:
We should fix that in results comparer. @adamsitnik I assume this was a "nice output" choice and not something technical?

Our actual reporting backend does does not truncate these names (@DrewScoggins to correct me). I'm in favor of descriptive names so we should do what we need to in order to make that possible.

Reply:
Made some renames and added comments.

Reply:
Yes, we don't truncate.